mera/mera_pymer4.py

```diff
@@ -159,6 +172,44 @@ def run_mera(
         else:
             print("WARNING: No data for IM, skipping...")
 
+        if verbose_warnings:
+            # Store the warning strings to print at the end
+            warning_str_per_im[cur_im]["stat_id"].extend(
+                utils.generate_insufficient_records_warning_str(
+                    cur_residual_df, "stat_id", min_num_records_per_site
+                )
+            )
+
+            warning_str_per_im[cur_im]["event_id"].extend(
+                utils.generate_insufficient_records_warning_str(
+                    cur_residual_df, "event_id", min_num_records_per_event
+                )
+            )
+
+    if verbose_warnings:
```

Comment:
Any particular reason why you store the warnings and then print them out, instead of just printing them out as they come up?

Reply:
There was a reason but I'm happy to discuss whether it's worth the extra code. If the warnings are printed as they come up, they are printed among the progress messages (such as "Processing IM pSA_8.697490026178, 117/118"). By storing the warnings, they can be printed at the end after all the progress messages have been printed. I thought this might make the warnings easier to notice but I'm not committed to keeping it this way because it does require a lot more code.

Reply:
I think it would have been perfectly fine to just print them out as they come up, but there is also nothing wrong with this either. So happy for you to leave it as is.